scala-thoughts.md

```diff
@@ -0,0 +1,14 @@
+This was my first time programming in Scala and it was definitely quite challenging. What was most challenging was finding the 
+Scala-y way of doign things. It was easy to resort to the imperative style of programming when I was unsure of how to approach
+a problem. Once I realized the pattern-matching capabilities of Scala, I ended up using it in a lot of the problems. The 
+functional style of programming gave me flashbacks to CS52 and SML. In terms of documentation, I found most of the resources 
+were a little too long-winded and not example-oriented enough, and it was annoying to have to frequently look at documentation
+but that's something that is part of the process of learning a new language. 
+
+It was also interesting to see the higher-order functions like filter, map and fold. Originally to write the sumOfMany
+function, I had tried to do some quite complex recursion, but then found the powerful function fold. These functions 
+seem to be both a blessing and a curse as they can drastically reduce the number of lines to accomplish something, 
+but lessens the need to understand the inner workings and intricacies of how things work. I can definitely see how it is a

```

Comment:
One thing I love about functional programming is getting to use the higher order functions you mentioned.  Pretty cool how powerful they are.